Specalc.package/Specalc.class/instance/equals.st

```diff
@@ -2,17 +2,7 @@ operations
 equals
 	buttonEq 
 		label: '=';
-		action: [ currentOperation = $+
-				ifTrue: [ display label: (((display label asNumber) + previousNumber) asString )] 
-				ifFalse: [ currentOperation = $-
-					ifTrue: [ display label: ((previousNumber - (display label asNumber)) asString ) ]
-					ifFalse: [ currentOperation = $x
-						ifTrue: [ display label: ((previousNumber *  (display label asNumber)) asString ) ]
-						ifFalse: [  (currentOperation = $/ & (display label asNumber)~= 0 )
-							ifTrue: [ display label: ((previousNumber / (display label asNumber)) asString ) ]
-							ifFalse: [ display label: 'Error' ]
-							]
-						 ]
-					]
-				];
-		state: true.
\ No newline at end of file
+		action: [ | addend total |
+			addend := display label asNumber.
+			total := previousNumber perform: currentOperation with: addend.
```

Comment:
This is the beauty of "everything is an object" to which we can pass messages. A little forethought and the operations themselves can be the message name of the operation (I only changed two - see below - so you could see the technique). And 13 lines become 3 clearly expressed lines!